backend/src/main/java/org/eclipse/tractusx/puris/backend/stock/domain/model/ProductStockRequestForMaterial.java

```diff
@@ -46,12 +42,14 @@ public class ProductStockRequestForMaterial {
 
     @NotNull
     @JsonProperty("materialNumberCustomer")
+    @Pattern(regexp = Material.MATERIAL_NUMBER_REGEX)
     private String materialNumberCustomer;
 
     @Pattern(regexp = "(^[0-9a-fA-F]{8}-[0-9a-fA-F]{4}-[0-9a-fA-F]{4}-[0-9a-fA-F]{4}-[0-9a-fA-F]{12}$)|(^urn:uuid:[0-9a-fA-F]{8}-[0-9a-fA-F]{4}-[0-9a-fA-F]{4}-[0-9a-fA-F]{4}-[0-9a-fA-F]{12}$)")
```

Comment:
This works. I propose to use the following as it's the same idea but simpler "(^urn:uuid:)?[0-9a-fA-F]{8}-[0-9a-fA-F]{4}-[0-9a-fA-F]{4}-[0-9a-fA-F]{4}-[0-9a-fA-F]{12}$"

I think the start character is not strictly necessary because the digits are limited to an exact length :)